Providers/Scripts/3.x/Scripts/nxOMSContainers.py

```diff
@@ -0,0 +1,295 @@
+
+    if stat_src.st_size != stat_dest.st_size:
+        return -1
+
+    if Checksum == "md5":
+        src_error = None
+        dest_error = None
+        src_hash = md5const()
+        dest_hash = md5const()
+        src_block = 'loopme'
+        dest_block = 'loopme'
+        src_file,src_error = opened_bin_w_error(SourcePath, 'rb')
+        if src_error:
+            Print("Exception opening source file " + SourcePath + " Error : " + str(src_error), file=sys.stderr)
+            LG().Log('ERROR', "Exception opening source file " + SourcePath + " Error : " + str(src_error))
+            return -1
+        dest_file, dest_error = opened_bin_w_error(DestinationPath, 'rb')
+        if dest_error:
+            Print("Exception opening destination file " + DestinationPath + " Error : " + str(dest_error), file=sys.stderr)
+            LG().Log('ERROR', "Exception opening destination file " + DestinationPath + " Error : " + str(dest_error))
+            src_file.close()
+            return -1
+        while src_block != '' and dest_block != '':
+            src_block = src_file.read(BLOCK_SIZE)
+            dest_block = dest_file.read(BLOCK_SIZE)
+            src_hash.update(src_block)
+            dest_hash.update(dest_block)
+            if src_hash.hexdigest() != dest_hash.hexdigest():
+                src_file.close()
+                dest_file.close()
+                return -1  
+        if src_hash.hexdigest() == dest_hash.hexdigest():
+            src_file.close()
+            dest_file.close()
+            return 0  
+    elif Checksum == "ctime":
+        if stat_src.st_ctime != stat_dest.st_ctime:
+            return -1
+        else:
+            return 0
+    elif Checksum == "mtime":
+        if stat_src.st_mtime != stat_dest.st_mtime:
+            return -1
+        else:
+            return 0
+
+def GetUID(User):
+    uid = None
+    try:
+        uid = pwd.getpwnam(User)[2]
+    except KeyError:
+        Print('ERROR: Unknown UID for ' + User, file=sys.stderr)
+        LG().Log('ERROR', 'ERROR: Unknown UID for ' + User)
+    return uid
+
+
+def GetGID(Group):
+    gid = None
+    try:
+        gid = grp.getgrnam(Group)[2]
+    except KeyError:
+        Print('ERROR: Unknown GID for ' + Group, file=sys.stderr)
+        LG().Log('ERROR', 'ERROR: Unknown GID for ' + Group)
+    return gid
+
+def StatFile(path):
+    """
+    Stat the file, following the symlink.
+    """
+    d = None
+    error = None
+    try:
+        d = os.stat(path)
+    except OSError as error:
+        Print("Exception stating file " + path  + " Error: " + str(error), file=sys.stderr)
+        LG().Log('ERROR', "Exception stating file " + path  + " Error: " + str(error))
+    except IOError as error:
+        Print("Exception stating file " + path  + " Error: " + str(error), file=sys.stderr)
+        LG().Log('ERROR', "Exception stating file " + path  + " Error: " + str(error))
+    return d
+    
+@contextmanager
+def opened_bin_w_error(filename, mode="rb"):
```

Comment:
Please note the differences of how this should be handled in the three different script versions of nxOMSPlugin:
2.4x-2.5x: https://github.com/Microsoft/PowerShell-DSC-for-Linux/blob/master/Providers/Scripts/2.4x-2.5x/Scripts/nxOMSPlugin.py#L407
2.6x-2.7x: https://github.com/Microsoft/PowerShell-DSC-for-Linux/blob/master/Providers/Scripts/2.6x-2.7x/Scripts/nxOMSPlugin.py#L407
3.x: https://github.com/Microsoft/PowerShell-DSC-for-Linux/blob/master/Providers/Scripts/3.x/Scripts/nxOMSPlugin.py#L418